contracts/DCAStrategies/DCAStrategies/DCAStrategiesPositionsHandler.sol

```diff
@@ -226,15 +250,98 @@ abstract contract DCAStrategiesPositionsHandler is IDCAStrategiesPositionsHandle
+    return !_isLastOne ? (_amount * _share) / _total : _amount - _amountSpent;
+  }
+
+  function _syncFirstBlock(
+    Position memory _position,
+    uint256 _totalAmount,
+    IDCAStrategies.ShareOfToken[] memory _newTokenShares,
+    uint32 _newAmountSwaps,
+    address _recipientSwapped
+  ) internal {
+    uint8 _currentPositionsIndex; // old positions index
+    uint8 _newPositionsIndex; // new positions index
+    uint256 _totalRemaining; // cash. The amount of money ready to use. (Used to know if need to send whats left or request whats missing)
+    uint256 _amountSpent;
+    Task[] memory _tasks = new Task[](_newTokenShares.length); // an array containing required tasks
+
+    // will iterate while arrays are not finished
+    while (_currentPositionsIndex < _position.positions.length && _newPositionsIndex < _newTokenShares.length) {
+      uint256 _currentPositionId = _position.positions[_currentPositionsIndex];
+      IDCAStrategies.ShareOfToken memory _newTokenShare = _newTokenShares[_newPositionsIndex];
+      IDCAHub.UserPosition memory _userPosition = _position.hub.userPosition(_currentPositionId);
+
+      uint256 _correspondingToPosition = _calculateOptimalAmount(
+        _newPositionsIndex == _newTokenShares.length - 1,
+        _totalAmount,
+        _newTokenShare.share,
+        _getTotalShares(),
+        _amountSpent
+      );
+      if (address(_userPosition.from) == _newTokenShare.token) {
+        // same token. Need to update position
+        if (_userPosition.remaining > _correspondingToPosition) {
+          // reduce
+          _syncReduceBlock(
+            _position.hub,
+            _tasks,
+            _newAmountSwaps,
+            _userPosition.remaining - _correspondingToPosition,
+            _currentPositionId,
+            _newPositionsIndex
+          );
+        } else if (_userPosition.remaining < _correspondingToPosition) {
+          // increase
+          _syncIncreaseBlock(_tasks, _correspondingToPosition - _userPosition.remaining, _currentPositionId, _newPositionsIndex);
+        }
+        _totalRemaining += _userPosition.remaining; // Do I need to add the remaining or just the diff between that and `_correspondingToPosition` (in the case of reduce)? (In the case of increase will be subtract the diff between `_correspondingToPosition` and `remaining`)
+        _newPositionsIndex++;
+        _currentPositionsIndex++;
+      } else if (_newTokenShare.token > address(_userPosition.from)) {
+        // then position needs to be deleted
+        _position.hub.terminate(_currentPositionId, address(this), _recipientSwapped);
+        _totalRemaining += _userPosition.remaining;
+        _currentPositionsIndex++;
+      } else {
+        // then just create a new position
+        _tasks[_newPositionsIndex] = Task({action: Action.DEPOSIT, positionId: 0, amount: _correspondingToPosition});
+        _newPositionsIndex++;
+      }
+    }
+  }
+
+  function _syncReduceBlock(
```

Comment:
I'm curious, why the extra function?

Reply:
`Stack too deep` lol

Reply:
But wait, I think I don't need that. Will try something